### PR TITLE
Navbar

### LINKED ISSUE #1: NavBar
The NavBar of the homepage. It should be according to what was discussed and can be viewed here:

https://docs.google.com/document/d/1DI9pc1EGDzOq2iV0QpOiCptKGl8OemCVOyguH4A94j0/edit#

Navigation links to the left <> SearchBar in the middle <> UserControl links to the right.

Should use a ready-to-go-component - search on the internet for a good one.

We might use react-bootstrap library (https://react-bootstrap.github.io/) so they might have something already.

### PR DESCRIPTION
**Pull request by arifur-ra**
I have done with navbar with few componets.like Logo,Usercontrol,Menubar,Search. please check the whole file and send me a review.

### CODE REVIEW

**Review by glaubernespoli — CHANGES_REQUESTED** [review]
- src/component/Header/Header.jsx => don't set the background color here (as in line 12). Checkout the latest version of the main (with the darkTheme applied) and it should use our theme throughout the app. Also the padding might be too much.
- src/component/Header/Logo/Logo.jsx => I really don't like putting styles directly like that..... it makes the code not standardized unless we copy paste this block of code everywhere. These things should be set in the themes file, unless it's something very specific (like maybe the logo icon).

- Header, Explore, MenuBar, Music, Search, UserControl: any reason you're wrapping everything around a <div> instead of a fragment (<>)?
- src/component/Header/Search/Search.jsx =>again, a lot of css set here. Maybe try to run everything after the dark theme has been applied and see if those are really necessary?

The rest looks great. Good job

File: src/component/Header/MenuBar/Explore.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import Button from '@material-ui/core/Button';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import * as React from 'react';
```
Comment (glaubernespoli) [inline src/component/Header/MenuBar/Explore.jsx]: dont import *... import only what you're gonna use (useState)

File: src/component/Header/MenuBar/MenuBar.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Button, Toolbar } from '@material-ui/core';
+import HomeIcon from '@material-ui/icons/Home';
+import React from 'react';
+import Explore from './Explore';
+import Music from './Music';
+
+const MenuBar = () => (
+  <div>
```
Comment (glaubernespoli) [inline src/component/Header/MenuBar/MenuBar.jsx]: any reason you keep wrapping everything around a `<div>` instead of a fragment `<>` ?

**Review by glaubernespoli — COMMENTED** [review]
Music and explore are exactly the same component with different information inside. It should be one component that receives the necessary props. Other than that code looks ok.

File: src/component/Header/MenuBar/MenuBar.jsx (hunk 2 above)
Comment (arifur-ra) [reply src/component/Header/MenuBar/MenuBar.jsx]: no reason for using div. i will do empty frangment.

File: src/component/Header/MenuBar/Explore.jsx (hunk 1 above)
Comment (arifur-ra) [reply src/component/Header/MenuBar/Explore.jsx]: okey, i will change it.

File: src/component/Header/MenuBar/MenuBar.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const classes = HeaderStyles();
+
+  return (
+    <>
+      <Toolbar>
+        <Button color="inherit" className={classes.logo}>
+          <HomeIcon />
+        </Button>
+        <NavItem
+          title="Music"
```
Comment (glaubernespoli) [inline src/component/Header/MenuBar/MenuBar.jsx:17]: don't make it like this, with subtitle, subtitle1, subtitle2... as this is limiting the amount of items we can have. If we decide we need more items later on, then we have to completely change the code. Just pass an array of subitems as the props

File: src/component/Header/MenuBar/NavItem.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'left'
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'left'
+        }}
+      >
+        <MenuItem onClick={handleClose}>{subTitle}</MenuItem>
```
Comment (glaubernespoli) [inline src/component/Header/MenuBar/NavItem.jsx]: Then here, with the subTitle array, you just call the map function and create a menuitem for each one of them

**Review by glaubernespoli — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -19,7 +19,7 @@
     "web-vitals": "^1.1.0"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "PORT=4000 react-scripts start",
```
Comment (glaubernespoli) [inline package.json]: dont change the port. revert this change

File: src/component/Header/MenuBar/NavItem.jsx [hunk 6]
```diff
@@ -41,17 +41,21 @@ const NavItem = ({ title, subTitle, subTitle1, subTitle2 }) => {
           horizontal: 'left'
         }}
       >
-        <MenuItem onClick={handleClose}>{subTitle}</MenuItem>
-        <MenuItem onClick={handleClose}>{subTitle1}</MenuItem>
-        <MenuItem onClick={handleClose}>{subTitle2}</MenuItem>
+        {subItems.map((subItem) => (
```
Comment (glaubernespoli) [inline src/component/Header/MenuBar/NavItem.jsx]: you dont need to declare a keyValue const to set the key. the map function (along with most array functions) can also receive the index and you can just use it as the key

**Review by glaubernespoli — CHANGES_REQUESTED** [review]

**Review by glaubernespoli — APPROVED** [review]
good!
